testdata/p4_16_samples_outputs/array-copy-bmv2-midend.p4

```diff
@@ -18,7 +18,8 @@ parser P(packet_in p, out Headers h, inout Meta m, inout standard_metadata_t sm)
     state start {
         p.extract<Hdr>(h.h1.next);
         p.extract<Hdr>(h.h1.next);
-        h.h2 = h.h1;
+        h.h2[0] = h.h1[0];
+        h.h2[1] = h.h1[1];
```

Comment:
There are cases where this kind of replacement is incorrect, because it does not preserve the nextIndex value of the header stack when one is assigned to the other.  We have at least one test case for that.  Is this incomplete/incorrect replacement being done in that case where it is incorrect, and caught as a failing test?  That test name is subparser-with-header-stack-bmv2.p4

Reply:
Perhaps your compiler changes are somehow checking, before making such a replacement, that the nextIndex value of the header stack on the LHS of the assignment statement is never used later?

Reply:
The pass already existed, I just enabled it. Assignment copies this invisible field?

Reply:
Assignment between header stacks is defined to copy the nextIndex field from one header stack to another, yes.

I checked, and this PR has no changes to the expected output files for test program subparser-with-header-stack-bmv2.p4, so perhaps these code changes are somehow only making the replacement when it is correct?  I do not know.